File: java-vertx/src/main/java/io/swagger/server/api/model/V3DirectionsResponse.java
package io.swagger.server.api.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import io.swagger.server.api.model.V3DirectionWithDescription;
import io.swagger.server.api.model.V3Status;
import java.util.ArrayList;
import java.util.List;
import io.vertx.core.json.JsonObject;
import io.vertx.codegen.annotations.DataObject;
/**
 * V3DirectionsResponse
 */

@javax.annotation.Generated(value = "io.swagger.codegen.v3.generators.java.JavaVertXServerCodegen", date = "2024-04-14T03:35:56.953023300+10:00[Australia/Sydney]")



public class V3DirectionsResponse   {
  
  private List<V3DirectionWithDescription> directions = null;

  private V3Status status = null;

  public V3DirectionsResponse directions(List<V3DirectionWithDescription> directions) {
    this.directions = directions;
    return this;
  }

  public V3DirectionsResponse addDirectionsItem(V3DirectionWithDescription directionsItem) {
    if (this.directions == null) {
      this.directions = new ArrayList<V3DirectionWithDescription>();
    }
    this.directions.add(directionsItem);
    return this;
  }

  /**
   * Directions of travel of route
   * @return directions
   **/
    public List<V3DirectionWithDescription> getDirections() {
    return directions;
  }

  public void setDirections(List<V3DirectionWithDescription> directions) {
    this.directions = directions;
  }

  public V3DirectionsResponse status(V3Status status) {
    this.status = status;
    return this;
  }

  /**
   * Get status
   * @return status
   **/
    public V3Status getStatus() {
    return status;
  }

  public void setStatus(V3Status status) {
    this.status = status;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    V3DirectionsResponse v3DirectionsResponse = (V3DirectionsResponse) o;
    return Objects.equals(this.directions, v3DirectionsResponse.directions) &&
        Objects.equals(this.status, v3DirectionsResponse.status);
  }

  @Override
  public int hashCode() {
    return Objects.hash(directions, status);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class V3DirectionsResponse {\n");
    
    sb.append("    directions: ").append(toIndentedString(directions)).append("\n");
    sb.append("    status: ").append(toIndentedString(status)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

File: dart/lib/model/v3_directions_response.dart
part of swagger.api;

class V3DirectionsResponse {
  /* Directions of travel of route */
  List<V3DirectionWithDescription> directions = [];

  V3Status status = null;

  V3DirectionsResponse();

  @override
  String toString() {
    return 'V3DirectionsResponse[directions=$directions, status=$status, ]';
  }

  V3DirectionsResponse.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    directions = V3DirectionWithDescription.listFromJson(json['directions']);
    status = new V3Status.fromJson(json['status']);
  }

  Map<String, dynamic> toJson() {
    return {
      'directions': directions,
      'status': status
     };
  }

  static List<V3DirectionsResponse> listFromJson(List<dynamic> json) {
    return json == null ? new List<V3DirectionsResponse>() : json.map((value) => new V3DirectionsResponse.fromJson(value)).toList();
  }

  static Map<String, V3DirectionsResponse> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, V3DirectionsResponse>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new V3DirectionsResponse.fromJson(value));
    }
    return map;
  }
}
